torch/_C/__init__.pyi.in

```diff
@@ -118,6 +119,14 @@ class _LegacyVariableBase(object):
     ) -> None: ...
 
 # Defined in torch/csrc/jit/python/init.cpp
+class Future(Generic[T]):
+  def __init__(self) -> None: ...
+  def wait(self) -> T: ...
+  def then(self, callback: Callable[[T], S]) -> Future[S]: ...
```

Comment:
Similar here, type of the first arg to Callable should be Future[T]?

Reply:
Decided against adding any generics to `torch/_C/__init.pyi`, because the actual pybind class does not inherit from Generic, which might cause further confusion.